contracts/UBI.sol

```diff
@@ -274,6 +298,59 @@ contract UBI is Initializable {
     emit Transfer(_account, address(0), _amount);
   }
 
+  /**
+  * @dev Approves, through a message signed by the `_owner`, `_spender` to spend `_value` tokens from `_owner`.
+  * @param _owner The address of the token owner.
+  * @param _spender The address of the spender.
+  * @param _value The amount of tokens to approve.
+  * @param _deadline The expiration time until which the signature will be considered valid.
+  * @param _v The signature v value.
+  * @param _r The signature r value.
+  * @param _s The signature s value.
+  */
+  function permit(address _owner, address _spender, uint256 _value, uint256 _deadline, uint8 _v, bytes32 _r, bytes32 _s) public {
+    require(_owner != address(0), "ERC20Permit: invalid owner");
```

Comment:
Should we add a `require(balanceOf(_spender) >= _value)` ?

Reply:
I guess you mean `require(balanceOf(_owner) >= _value)`, aren't you?

I think we shouldn't. The `permit` is an alternative to perform an `approve`, which allows you to modify the allowance to whatever value you choose. I think that's useful for a lot of use cases because allows you to approve the spending of tokens you don't own yet.